app/controllers/commodities_controller.rb

```diff
@@ -23,17 +22,43 @@ def commodities_by_code
     Commodity.by_code(search_term).sort_by(&:code)
   end
 
+  def fetch_headings_or_commodities
+    if heading_from_param&.declarable?
+      fetch_headings
+    else
+      fetch_commodities
+    end
+  end
```

Comment:
I added this to avoid adding another conditional before_action filter, and try to be more explicit.